main.py

```diff
@@ -51,6 +73,64 @@ def get(self):
         template = JINJA_ENVIRONMENT.get_template('templates/auth.html')
         self.response.write(template.render({}))
 
+def FilterItems():
+    if CostumeFilterEnabled() == True and PropFilterEnabled() == False:
+        query = Item.query(Item.type == "Costume").order(-Item.updated, Item.condition)
+    elif CostumeFilterEnabled() == False and PropFilterEnabled() == True:
+        query = Item.query(Item.type == "Prop").order(-Item.updated, Item.condition)
+    else:
+        query = Item.query().order(-Item.updated, Item.condition)
+
+    # Handles Condition selection options
+    if ConditionGoodFilterEnabled():
+        if ConditionFairFilterEnabled():
+            if ConditionPoorFilterEnabled():
+                if ConditionRepairFilterEnabled():
+                    query = query.filter(ndb.OR(Item.condition == "Good", Item.condition == "Fair", Item.condition == "Poor", Item.condition == "Being Repaired"))
```

Comment:
I only stumbled across this during the summer, but there is a nifty IN operator hidden away in the ndb documentation (https://cloud.google.com/appengine/docs/standard/python/ndb/queries#neq_and_in) so you could do something like 

```py
query(Item.condition` IN ["Good", "Fair", "Poor", "Being Repaired"]) 
```

This is nice since it can make the code smaller.

```py
conditions_filter = []
if ConditionGoodFilterEnabled():
    conditions.append("Good")
if ConditionFairFilterEnabled():
    conditions.append("Fair")
if ConditionPoorFilterEnabled():
    conditions.append("Poor")
if ConditionRepairFilterEnabled():
    conditions.append("Being Repaired")

query = query.filter(Item.condition IN conditions_filter)
```

You may even be able to create the filter directly from the check boxes rather than the function calls by doing something like here (https://stackoverflow.com/questions/13242412/how-to-take-checkboxes-in-python) where they use the same name, different values, and request.get_all rather than request.get. This would require some small changes to the HTML but I think it could look a little cleaner.

Reply:
I dabbled with IN and didn't get very far, but I think I know how to make it work now. I agree that the if statements are very large and I'll try shrinking them with IN.